soh/soh/ActorDB.h

```diff
@@ -0,0 +1,79 @@
+#pragma once
+#include "z64actor.h"
+
+typedef struct {
+    const char* name;
+    const char* desc;
+    u32 valid;
+    s32 id;
+    s32 category;
+    u32 flags;
+    s32 objectId;
+    size_t instanceSize;
+    ActorFunc init;
+    ActorFunc destroy;
+    ActorFunc update;
+    ActorFunc draw;
+    ActorResetFunc reset;
+    s32 numLoaded;
+} ActorDBEntry;
+
+#ifdef __cplusplus
+
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+struct ActorDBInit {
+    std::string name;
+    std::string desc;
+    int category = 0;
+    u32 flags = 0;
+    int objectId = 0;
+    size_t instanceSize = sizeof(Actor);
+    ActorFunc init = nullptr;
+    ActorFunc destroy = nullptr;
+    ActorFunc update = nullptr;
+    ActorFunc draw = nullptr;
+    ActorResetFunc reset = nullptr;
+};
+
+class ActorDB {
+public:
+    static ActorDB* Instance;
+
+    ActorDB();
+
+    // Wrapper around ActorDBEntry so we get C++isms for the entries
+    struct Entry {
+        Entry();
+        Entry(const Entry& other);
+        Entry& operator=(const Entry& other);
+
+        void setName(const std::string& newName);
+        void setDesc(const std::string& newDesc);
+
+        std::string name;
+        std::string desc;
+        ActorDBEntry entry;
+    };
+    Entry& Add(const ActorDBInit& init);
+
+    Entry& Lookup(const int id);
+    int LookupId(const std::string& name);
```

Comment:
I'm partial to the `Retrieve` naming scheme as opposed to `Lookup`, like I've done with CustomMessages and GetItemEntry tables already, but if @briaguya-ai or other people like `Lookup` more then I'll want to change my systems to use that as well. Either way I definitely think all the modern replacements for legacy tables should use the same naming scheme (makes for a better modding API down the road imo).

Reply:
Agreed for the naming scheme. It should be consistent since both system have the same idea. I'm open to whatever the admins feel like.

Reply:
i don't have strong opinions one way or the other on the naming, i have gotten used to `Retrieve` though so my gut feeling is stick with that